Sources/AsyncHTTPClient/Response+Decodable.swift

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Swift Server Working Group and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import NIO
+
+
+extension EventLoopFuture where Value == HTTPClient.Response {
+    
+    public enum BodyError : Swift.Error {
+        case noBodyData
+    }
+    
+    /// Decode the response body as T using the given decoder.
+    ///
+    /// - parameters:
+    ///     - type: The type to decode.  Must conform to Decoable.
+    ///     - decoder: The decoder used to decode the reponse body.  Defaults to JSONDecoder.
+    /// - returns: A future decoded type.
+    /// - throws: BodyError.noBodyData when no body is found in reponse.
+    public func decode<T : Decodable>(as type: T.Type, decoder: Decoder = JSONDecoder()) -> EventLoopFuture<T> {
```

Comment:
Is there any reason these functions need to be extensions on `EventLoopFuture` instead of on `HTTPClient.Response`?

Reply:
This is simply convenience to allow the user to do something like this. 
`httpClient.get(url: url).decode(as: Something.self)`

Reply:
Yeah, I think I'd rather have the visual noise of requiring that this appear in a `map`.

Reply:
Ok.  Whats the thinking with that decision?  The only reason I even thought to create this extension was because I found myself ~~writing~~ copy/pasting the same boilerplate map function to all of my requests.  I can see cases where one would want a one-off, custom decode handler but that seems like a poor reason to require it to be written out ever time.